document-store/src/main/java/org/hypertrace/core/documentstore/query/Query.java

```diff
@@ -205,6 +205,11 @@ public QueryBuilder setFilter(final FilterTypeExpression expression) {
       return this;
     }
 
+    public QueryBuilder clearFilter() {
```

Comment:
The clients would never want this here. Maybe, we could move this method inside the TransformedQueryBuilder instead?

Reply:
I was thinking about it. But in `TransformedQueryBuilder` we need a different way.

Reply:
Providing support in TransformedQueryBuilder by exposing copy functions.

Reply:
Done.